### ISSUE TITLE
flow-capture init script doesn't do anything - Redhat 6.9

### ISSUE THREAD

**Issue by hackdefendr** [Cacti/plugin_flowview#7]
Attempted replace the flow-tools init script with the optional one included with Flowview.  However, there is no listening port opened nor is there any processes started when using the included init script.

Verified that when I use the OS installed init script, there is a listening port created, and a process is started. Additionally, I see flows being generated in the data folder.  

Basically, nothing in this plugin works.

**Comment by cigamit** [comment]
The most complicated part of setting up this plugin is properly configuring your Listener.  Otherwise it works well.  The init script provide by this plugin is also required and not the default one from the flow tools.

**Comment by cigamit** [comment]
Here is an example Listener configured and working.  Next I'll show the running script.

![examplelistener](https://cloud.githubusercontent.com/assets/957322/25525501/cc3f8468-2bd4-11e7-9f9a-edf4ea4db41c.PNG)

**Comment by cigamit** [comment]
Corresponding capture syntax:

![flowcapture](https://cloud.githubusercontent.com/assets/957322/25525570/2a1e343a-2bd5-11e7-8918-e80172e5ca43.PNG)

**Comment by cigamit** [comment]
Head of the init script in /etc/init.d.  Seems to me a nice product would be to migrate this script to systemd.  Someone will have to do that as well as to move the charts from OpenFlashCharts to C3.js.

![flowcapturescript](https://cloud.githubusercontent.com/assets/957322/25525704/c7592d4a-2bd5-11e7-8a45-ab6e392faa42.PNG)

**Comment by cigamit** [comment]
So, I was hacking around and found that the init script was missing a very important variable declaration.  I've subsequently updated it.  You should update and then re-implement it.  Don't forget to follow the instructions.  If you find issues and would like to enhance the instructions, please create a pull request.  Thanks!

**Comment by hackdefendr** [comment]
After updating, I am seeing this error in the Cacti Logs now:
`2017-04-28 08:21:44 - CMDPHP PHP ERROR WARNING Backtrace: (/etc/rc.d/init.d/flow-capture: 20 start)(/etc/rc.d/init.d/flow-capture: 57 mkdir)(CactiErrorHandler)(/lib/functions.php: 4296 cacti_debug_backtrace)
2017-04-28 08:21:44 - ERROR PHP WARNING: mkdir(): No such file or directory in file: /etc/rc.d/init.d/flow-capture on line: 57 `

I looked at line 57 of the init script and did not see anything wrong.  My flows will be stored in /data/flows and that folder does exist. So I am a bit stumped on this error.

**Comment by hackdefendr** [comment]
OK so as a test...I removed the mkdir section and just left the part of code that starts flow-capture.  This works as long as the folder structure is already created. Not sure why the script is failing to make directories.

So in this hacked state the init script does work and starts flow-capture properly.  Obviously not ideal, but it will suffice for my testing.

**Comment by cigamit** [comment]
Did the base path not exist or just the device folder?  We may have to change to using a recursive create like this
mkdir("$path/$folder", 0777, true);
If the base path did exist, then it could be permissions or SELinux, etc...

**Comment by hackdefendr** [comment]
Initially the folders existed, but I deleted them to see if it would be created. This didn't work, so I manually created the folders, and realized the folder it was trying to create was being pulled from my listener config, which was still set to /data/flows instead of localhost. I thought maybe that would solve it, but it still errored out.  I will restore the init script and try your suggestion with the 0777 bit and report back if that works or not.
